l1-contracts/contracts/bridge/asset-tracker/AssetTracker.sol

```diff
@@ -144,6 +194,68 @@ contract AssetTracker is IAssetTracker, Ownable2StepUpgradeable, AssetHandlerMod
         chainBalance[chainToUpdate][_assetId] -= _amount;
     }
 
+    function handleChainBalanceIncreaseOnGateway(
+        uint256 _chainId,
+        bytes32 _canonicalTxHash,
+        bytes32 _baseTokenAssetId,
+        uint256 _baseTokenAmount,
+        bytes32 _assetId,
+        uint256 _amount
+    ) external {
+        if (_amount > 0) {
+            chainBalance[_chainId][_assetId] += _amount;
+        }
+        if (_baseTokenAmount > 0) {
+            chainBalance[_chainId][_baseTokenAssetId] += _baseTokenAmount;
+        }
+        balanceChange[_chainId][_canonicalTxHash] = BalanceChange({
```

Comment:
do we care about colissions in `canonicalTxHash` (only possible in case of a bad CTM)? Probably not, in this case, it is better to leave a comment on why

Reply:
added comment